src/Strategy/Python/Poetry/Common.hs

```diff
@@ -1,65 +1,84 @@
-{-# LANGUAGE RecordWildCards #-}
-
 module Strategy.Python.Poetry.Common (
   PoetryProject (..),
-  buildPyProjectGraph,
   buildGraphWithLock,
   buildPackageNameGraph,
   discover,
+  toMap,
 ) where
 
 import Control.Effect.Diagnostics (Diagnostics, context)
-import Data.Map.Strict qualified as M
+import Data.Foldable (asum)
+import Data.Map.Strict qualified as Map
 import Data.Maybe (fromMaybe)
-import Data.Text (toLower)
-import DepTypes
+import Data.Text (Text, toLower)
+import DepTypes (DepEnvironment (..), DepType (..), Dependency (..), VerConstraint (..))
 import Discovery.Walk (
   WalkStep (WalkContinue, WalkSkipAll),
   findFileNamed,
   walk',
  )
-import Effect.Exec
+import Effect.Exec (Has)
```

Comment:
[nit]: `Has` originally comes from `Control.Algebra`, we just re-export it from all of the effect modules.  Importing an item from a module that reexports it is fine, **as long as you also use something else in that module**, otherwise you're adding an unnecessary edge to the build graph that cabal must build.

My suggestion is to import `Has` from either `Control.Effect.Diagnostics` since you already use that, or directly from `Control.Algebra`, so there's no confusion if we ever modify the effects in this file.

```suggestion
import Control.Algrbra (Has)
```